static/js/subs.js

```diff
@@ -923,10 +923,16 @@ exports.change_stream_name = function (e) {
             new_name_box.val('');
             ui.report_success(i18n.t("The stream has been renamed!"), $("#subscriptions-status "),
                               'subscriptions-status');
+            $("#name_change_error_" + stream_id).hide();
         },
         error: function (xhr) {
             ui.report_error(i18n.t("Error renaming stream"), xhr,
                             $("#subscriptions-status"), 'subscriptions-status');
+            new_name_box.attr("contenteditable", "true").focus();
+            $(e.target).parent().find(".checkmark").show();
+            $(e.target).parent().find(".editable").text("×");
+            $("#name_change_error_" + stream_id).text(i18n.t("A stream with this name already exists!"));
```

Comment:
The `xhr` passed back has an error message in it, and you should display that rather than a fixed error.

E.g. another common error condition is a too-long name, which should be displayed differently.  

You can likely clean up this code by using `ui.report_error(), which takes care of doing the extraction from an xhr for errors with a clear error message in the xhr.

The other UI issue I see is that after getting an error that the stream name is taken, if you hit the "x" to cancel, it should hide the `name_change_error`.